qubes/tests/integ/storage.py

```diff
@@ -257,6 +257,30 @@ async def _test_003_snapshot(self):
                     size),
                 user='root')
 
+        await self.vm2.shutdown(wait=True)
```

Comment:
This shutdown consistently times out. VM log ends with `A stop job is running for Session c3 of User user` (which I think is the one that causes the timeout). Maybe some weird interaction with shutting down when not everything started yet? Try adding `wait_for_session(self.vm2)` before.

Reply:
> This shutdown consistently times out. VM log ends with `A stop job is running for Session c3 of User user` (which I think is the one that causes the timeout).

After resetting my `default_shutdown_timeout` (I had it cranked up to 300 seconds) I saw the same kind of error but for vm1.

> Maybe some weird interaction with shutting down when not everything started yet? Try adding `wait_for_session(self.vm2)` before.

Waiting for sessions did the trick. There even was a convenient `start_vm()` helper method.